src/Components/src/Microsoft.AspNetCore.Components.Server/Builder/RazorComponentsApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Components.Server;
+using Microsoft.AspNetCore.Components.Server.Builder;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    /// <summary>
+    /// Extension methods to configure an <see cref="IApplicationBuilder"/> for serving interactive components.
+    /// </summary>
+    public static class RazorComponentsApplicationBuilderExtensions
+    {
+        /// <summary>
+        /// Adds middleware for serving interactive Razor Components.
+        /// </summary>
+        /// <param name="builder">The <see cref="IApplicationBuilder"/>.</param>
+        /// <typeparam name="TStartup">A components app startup type.</typeparam>
+        /// <returns>The <see cref="IApplicationBuilder"/>.</returns>
+        public static IApplicationBuilder UseRazorComponents<TStartup>(
```

Comment:
Technically we don't need the `TStartup` at all. I've left it here for symmetry with `AddRazorComponents` and in case we want to reserve the right to use that info in the future.

I'm totally open to changing this though.

Reply:
Good to know. I think we should leave it as-is until we resolve all of the dependencies and tweaks needed to ship preview2.